File: CR.Elastos-Nucleus/service/sample_code/python/generate_key.py
import json

from elastos_adenine.common import Common

shared_secret_adenine = "7XDnFBdHafpPyIC4nrtuJ5EUYVqdEKjW"

def main():
    host = "localhost"
    port = 8001
    production = False

    did = "iHdasfhasdflkHdasfasdfD"

    generate_api_key(host, port, production, did)
    get_api_key(host, port, production, did)

def generate_api_key(host, port, production, did_to_use):
    try:
        common = Common(host, port, production)
        # Generate API Key
        print("--> Generate API Key - SHARED_SECRET_ADENINE")
        response = common.generate_api_request(shared_secret_adenine, did_to_use)
        if response['status']:
            json_output = json.loads(response['output'])
            print("Status Message :", response['status_message'])
            for i in json_output['result']:
                print(i, ':', json_output['result'][i])
        else:
            print("Error Message: " + response['status_message'])
    except Exception as e:
        print(e)
    finally:
        common.close()

def get_api_key(host, port, production, did_to_use):
    try:
        common = Common(host, port, production)
        # Get API Key
        print("--> Get API Key - SHARED_SECRET_ADENINE")
        response = common.get_api_key_request(shared_secret_adenine, did_to_use)
        if response['status']:
            json_output = json.loads(response['output'])
            print("Status Message :", response['status_message'])
            for i in json_output['result']:
                print(i, ':', json_output['result'][i])
        else:
            print("Error Message: " + response['status_message'])
    except Exception as e:
        print(e)
    finally:
        common.close()

if __name__ == '__main__':
    main()

File: CR.Elastos-Nucleus/service/sample_code/go/generate_key.go
package main

import (
	"log"

	"github.com/cyber-republic/go-grpc-adenine/elastosadenine"
)

const (
    sharedSecretAdenine string = "7XDnFBdHafpPyIC4nrtuJ5EUYVqdEKjW"
)

func main() {
    grpcServerHost := "localhost"
    grpcServerPort := 8001
    production := false
    
    did := "iHdasfhasdflkHdasfasdfD"

    generateAPIKeyDemo(grpcServerHost, grpcServerPort, production, did)
    getAPIKeyDemo(grpcServerHost, grpcServerPort, production, did)
}

func generateAPIKeyDemo(grpcServerHost string, grpcServerPort int, production bool, didToUse string) {
	common := elastosadenine.NewCommon(grpcServerHost, grpcServerPort, production)
	defer common.Close()

	log.Println("--> Generate API Key - SHARED_SECRET_ADENINE")
	responseSSAdenine := common.GenerateAPIRequest(sharedSecretAdenine, didToUse)
	if responseSSAdenine.Status {
		log.Printf("Output: %s", responseSSAdenine.Output)
	} else {
		log.Printf("Error Message: %s", responseSSAdenine.StatusMessage)
	}
}

func getAPIKeyDemo(grpcServerHost string, grpcServerPort int, production bool, didToUse string) {
	common := elastosadenine.NewCommon(grpcServerHost, grpcServerPort, production)
	defer common.Close()

	log.Println("--> Get API Key - SHARED_SECRET_ADENINE")
	responseSSAdenine := common.GetAPIKey(sharedSecretAdenine, didToUse)
	if responseSSAdenine.Status {
		log.Printf("Output: %s", responseSSAdenine.Output)
	} else {
		log.Printf("Error Message: %s", responseSSAdenine.StatusMessage)
	}
}

